torchgeo/transforms/temporal.py

```diff
@@ -68,3 +71,29 @@ def compute_transformation(
         """
         out = self.identity_matrix(input)
         return out
+
+
+class TemporalEmbedding(nn.Module):
+    """Generic sinusoidal embedding for periodic temporal features."""
+
+    def __init__(self, period: int) -> None:
+        """Initialize a TemporalEmbedding instance.
+
+        Args:
+            period: The period of the sinusoidal function.
+        """
+        super().__init__()
+        self.period = period
+
+    def forward(self, t: Tensor) -> Tensor:
+        """Compute sinusoidal embeddings for periodic time values.
+
+        Args:
+            t: Tensor of shape (B,) or (B, 1), representing time values.
+
+        Returns:
+            Tensor of shape (B, 2), sin and cos embeddings.
+        """
+        t = t.view(-1, 1).float()
```

Comment:
Don't need to rearrange?

Convert t from float to pd.Timestamp before division, then convert back to float after division?